File: frontend/src/types/body.ts
export interface Body {
  id: number;
  simulationId: number;
  mass: number;
  initialPosition: { x: number; y: number };
  initialVelocity: { x: number; y: number };
  modelPath: string;
  createdAt: Date;
  updatedAt: Date;
}

export interface BodyState {
  bodyId: number;
  positionX: number;
  positionY: number;
  velocityX: number;
  velocityY: number;
}

File: backend/api_server/src/models/body.py
from datetime import datetime

from pydantic import BaseModel


class Vec2(BaseModel):
    x: float
    y: float


class Body(BaseModel):
    id: int
    simulation_id: int
    mass: float
    initial_position: Vec2
    initial_velocity: Vec2
    model_path: str
    created_at: datetime
    updated_at: datetime
